R/prep_lintr.R

```diff
@@ -5,10 +5,20 @@ linters_to_lint <- list(
   assignment_linter = lintr::assignment_linter(),
   line_length_linter = lintr::line_length_linter(80),
   trailing_semicolon_linter = trailing_semicolon_linter(),
-  attach_detach_linter = attach_detach_linter(),
-  setwd_linter = setwd_linter(),
-  sapply_linter = sapply_linter(),
-  library_require_linter = library_require_linter(),
+  attach_detach_linter = lintr::undesirable_function_linter(fun = c(
+    "attach" = "Avoid attach, it is easy to create errors with it",
+    "detach" = "Avoid detach, it is easy to create errors with it"
+  )),
+  setwd_linter = lintr::undesirable_function_linter(fun = c(
```

Comment:
I think you might be able to rely on the defaults https://lintr.r-lib.org/reference/default_undesirable_functions.html

Reply:
Yeah, I pondered that, and it is of course possible, but the current setup has this useful explicit list here:
https://github.com/ropensci-review-tools/goodpractice/blob/c3b60fb3f6be4ffae2a835631cb27119f166db12/R/prep_lintr.R#L4-L13
Bundling all in single default `undesirable_functions` would then require people to look somewhere in `lintr` to find out which default functions were considered undesirable. I like this explicit naming more. Does that make sense?

Reply:
But would there be a way for us not to store the messages here? Like for instance having `"Avoid changing the working directory, or restore it in on.exit"` in goodpractice source.

Reply:
Yes, of course there is! Thank you so much for pushing further. Code in current form is much neater and more robust. thanks!